pkg/reaper/nodereaper/nodereaper.go

```diff
@@ -289,8 +327,85 @@ func (ctx *ReaperContext) deriveFlappyDrainReapableNodes() error {
 	return nil
 }
 
+// Handle ghost-reapable nodes
+func (ctx *ReaperContext) deriveGhostDrainReapableNodes(w ReaperAwsAuth) error {
+	log.Infoln("scanning for ghost drain-reapable nodes")
+	for instance, node := range ctx.NodeInstanceIDs {
```

Comment:
This is going to make 2 AWS API calls for every node in the cluster. On large/busy clusters, the AWS API calls might start getting throttled. Is there a way to optimize this?

Make 2 AWS API calls to get ALL "Terminated instances" in the cluster (with the KubernetesCluster tag) and check the output against all nodes in the cluster. Would this work?

Reply:
You're right, I can probably make a single Describe call before the loop, and do implement the filtering in governor.
It's a bit more code, but for good cause

Reply:
Fixed by caching all instances and implementing filtering